index-providers/modeshape-elasticsearch-index-provider/src/test/java/org/modeshape/jcr/index/elasticsearch/EsIndexTest.java

```diff
@@ -0,0 +1,369 @@
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.node.NodeBuilder;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+import org.modeshape.common.util.FileUtil;
+import org.modeshape.jcr.ExecutionContext;
+import org.modeshape.jcr.api.query.qom.Operator;
+import org.modeshape.jcr.cache.NodeKey;
+import org.modeshape.jcr.index.elasticsearch.client.EsClient;
+import org.modeshape.jcr.query.model.And;
+import org.modeshape.jcr.query.model.Between;
+import org.modeshape.jcr.query.model.Comparison;
+import org.modeshape.jcr.query.model.FullTextSearch;
+import org.modeshape.jcr.query.model.Length;
+import org.modeshape.jcr.query.model.Literal;
+import org.modeshape.jcr.query.model.LowerCase;
+import org.modeshape.jcr.query.model.Not;
+import org.modeshape.jcr.query.model.Or;
+import org.modeshape.jcr.query.model.PropertyExistence;
+import org.modeshape.jcr.query.model.PropertyValue;
+import org.modeshape.jcr.query.model.SelectorName;
+import org.modeshape.jcr.query.model.SetCriteria;
+import org.modeshape.jcr.query.model.UpperCase;
+import org.modeshape.jcr.spi.index.IndexConstraints;
+import org.modeshape.jcr.spi.index.ResultWriter;
+import org.modeshape.jcr.spi.index.provider.Filter.Results;
+import org.modeshape.jcr.value.Name;
+import org.modeshape.jcr.value.ValueFactories;
+
+/**
+ *
+ * @author kulikov
+ */
+public class EsIndexTest {
+
+    private final static ExecutionContext context = new ExecutionContext();
+    private final static EsIndexColumn def1 = new EsIndexColumn(context, "field1", PropertyType.STRING);
+    private final static EsIndexColumn def2 = new EsIndexColumn(context, "field2", PropertyType.DECIMAL);
+    private final static EsIndexColumn def3 = new EsIndexColumn(context, "field3", PropertyType.STRING);
+    private final static EsIndexColumn def4 = new EsIndexColumn(context, "mixinTypes", PropertyType.NAME);
+    private final static EsIndexColumn def5 = new EsIndexColumn(context, "myfield", PropertyType.STRING);
+    private static Node esNode;
+    private static EsClient client;
+    private static EsIndex index;
+
+    public EsIndexTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() {
+        FileUtil.delete("target/data");
+
+        EsIndexColumns cols = new EsIndexColumns(def1, def2, def3, def4, def5);
+        esNode = NodeBuilder.nodeBuilder().local(false)
+                .settings(Settings.settingsBuilder().put("path.home", "target/data"))
+                .node();
+        client = new EsClient("localhost", 9200);
+        index = new EsIndex(client, cols, context, "test", "workspace");
+
+        initialIndex();
+        
+        try {
+            Thread.currentThread().sleep(1000);
+        } catch (Exception e) {
+        }
```

Comment:
the only possible checked exception here is `InterruptedException` which if thrown, should be handled accordingly (i.e. interrupted status cleared and perhaps rethrown)